### ISSUE TITLE
Thread-safety issue in TotemTransport

### ISSUE THREAD

**Issue by makortel** [cms-sw/cmssw#36544]
Investigation in #35109 pointed to this code
https://github.com/cms-sw/cmssw/blob/a9f44839b30cab42562af4434e1f01c8b1af8cc6/SimTransport/PPSProtonTransport/src/TotemTransport.cc#L82-L87
that, when called, modifies an existing event data product (violates policy and is not thread safe)
https://github.com/cms-sw/cmssw/blob/a9f44839b30cab42562af4434e1f01c8b1af8cc6/SimPPS/PPSSimTrackProducer/plugins/PPSSimTrackProducer.cc#L117-L118
https://github.com/cms-sw/cmssw/blob/a9f44839b30cab42562af4434e1f01c8b1af8cc6/SimPPS/PPSSimTrackProducer/plugins/PPSSimTrackProducer.cc#L131

While apparently this code is not being called in runTheMatrix workflows, it should be fixed.

**Comment by makortel** [comment]
assign simulation

**Comment by makortel** [comment]
FYI @cms-sw/ctpps-dpg-l2

**Comment by civanch** [comment]
I will try to fix...

**Comment by civanch** [comment]
@mundim, @fabferro , 
#36589 should fix thread safety, however, PPS simulation is still disabled by default, because was breaking thread safety. When #36589 will be in IB,  please, try to perform new tests with PPS enabled in g4SimHits.

There is also a question: if PPS is enabled are parameters of a proton the same as given by PYTHIA or they are resampled in BaseProtoneTransport?

**Comment by perrotta** [comment]
Even if #36589, which includes the proposed fixes, I don't think this issue can get closed until @cms-sw/ctpps-dpg-l2 or @cms-sw/proton-pog-l2 explicitely verify it by enabling proton transport in the code (or until proton transport is re-enabled by default in CMSSW and we can verify that there are no more thread safety related isues)

**Comment by civanch** [comment]
I agree that what is fixed is the only one problem and we may have more. After the new IB I will try to enable PPS and we can see if it is working or not.

**Comment by civanch** [comment]
+1

it seems that the problem does not show up for few weeks after the fix.
